src/datachain/lib/signal_schema.py

```diff
@@ -749,3 +749,120 @@ def _build_tree_for_model(
             res[name] = (anno, subtree)  # type: ignore[assignment]
 
         return res
+
+    def to_partial(self, *columns: str) -> "SignalSchema":
+        """
+        Convert the schema to a partial schema with only the specified columns.
+
+        E.g. if original schema is:
+
+            ```
+            signal: Foo@v1
+                name: str
+                value: float
+            count: int
+            ```
+
+        Then `to_partial("signal.name", "count")` will return a partial schema:
+
+            ```
+            signal: FooPartial@v1
+                name: str
+            count: int
+            ```
+
+        Note that partial schema will have a different name for the custom types
+        (e.g. `FooPartial@v1` instead of `Foo@v1`) to avoid conflicts
+        with the original schema.
+
+        Args:
+            *columns (str): The columns to include in the partial schema.
+
+        Returns:
+            SignalSchema: The new partial schema.
+        """
+        serialized = self.serialize()
+        custom_types = serialized.get("_custom_types", {})
+
+        schema: dict[str, Any] = {"_custom_types": {}}
+
+        for column in columns:
+            parent_type_origin, parent_type_partial = "", ""
+            for i, signal in enumerate(column.split(".")):
+                # We are expecting for the first part of the column name (signal name)
+                # to be in the schema, other parts should be in the custom types.
+                if i == 0:
```

Comment:
can be a column with a custom type w/o any additional parts `bbox`

in this case it seems we won't copy `_custom_type`, right?

Reply:
Not sure 🤔
Added one more test for our custom model (`BBox`), but still not sure if this is what you have meant.

Reply:
what are the possible inputs to this function? Can I pass `bbox` (and expect all nested fields included), for example? Or is always the most granular level nested fields (and reconstruct parents from them)?

Reply:
Now I see. Good question. It is supposed to always take the most granular level nested fields, but since this function is exposed, anyone can pass, let's say, `bbox`. Let me check and add new test for this.

Reply:
I think that's fine, can be an assert I guess - if we don't need a general helper let's keep it scoped, I would clarify the expected inputs though